pkg/matcher/matcher.go

```diff
@@ -39,6 +66,33 @@ func (w *allowlistMatcher) Match(fullPath string, _ os.FileInfo, _ io.ReaderAt)
 	return false, false
 }
 
+func (w *allowlistMatcher) GetCommonPrefixDirs() []string {
+	return findCommonDirPrefixes(w.allowlist)
+}
+
+func findCommonDirPrefixes(prefixes []string) []string {
+	pre := make(map[string]set.StringSet)
+	for _, d := range prefixes {
+		for d != "" {
+			p, _ := path.Split(strings.TrimSuffix(d, "/"))
+			s := pre[p]
+			s.Add(d)
+			pre[p] = s
+			d = p
+		}
+	}
+	dirs := pre[""].AsSlice()
+	ret := dirs[:0]
```

Comment:
It looks like the common dir prefix just steps one level down from the root directory and it returns exactly one common prefix per first level dir referenced. And yes, that could be a good balance for how common the common prefix could be. Can you add more comments?
You can also rename the dirs as firstLevelDirs or dirsUnderRoot to make it more clear.

Reply:
:+1: